pilot/pkg/bootstrap/server.go

```diff
@@ -138,7 +141,7 @@ type Server struct {
 }
 
 // NewServer creates a new Server instance based on the provided arguments.
-func NewServer(args *PilotArgs) (*Server, error) {
+func NewServer(args *PilotArgs, stopCh <-chan struct{}) (*Server, error) {
```

Comment:
This is not right. stopCh is never actually used. I think we need to use the same stopCh that is used for startFunc which may mean we do everything inside startFuncs

Reply:
Yes, basically the stop channel in the Server is only for test.

Reply:
Per discussion, the fix has been added to pkg/test/framework/components/pilot/native.go.